src/com/amazon/corretto/crypto/provider/MlKemGen.java

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.KeyPair;
+import java.security.KeyPairGeneratorSpi;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+
+class MlKemGen extends KeyPairGeneratorSpi {
+  private final AmazonCorrettoCryptoProvider provider_;
+  private int parameterSet = -1;
```

Comment:
Same comment as elsewhere, use a Java enum for the parameter size, not a raw int or Integer.